extensions/amp-story-interactive/1.0/amp-story-interactive-abstract.js

```diff
@@ -238,7 +243,23 @@ export class AmpStoryInteractive extends AMP.BaseElement {
    * @public
    */
   initializeState() {
-    this.updateStoryStoreState_(null);
+    this.variableService_ = Services.storyVariableService(this.win);
+    Promise.all([
+      Services.storyStoreServiceForOrNull(this.win).then((storeService) => {
+        this.storeService_ = storeService;
+        this.updateStoryStoreState_(null);
+      }),
+      Services.storyRequestServiceForOrNull(this.win).then((requestService) => {
+        this.requestService_ = requestService;
+      }),
+      Services.storyAnalyticsServiceForOrNull(this.win).then(
+        (analyticsService) => {
+          this.analyticsService_ = analyticsService;
+        }
+      ),
+    ]).then(() => {
```

Comment:
1. This should all happen in the build or layoutCallback which can be async (cf Story codelab), so you don't need to `initializePromiseResolve_` at all. This extra layer of async code makes things more complex when it's actually solved by the async build/layoutCallback.

2. `this.updateStoryStoreState_(null);` you could extract what this does to a helper method (computing the ID and sending it to the store?) and call it from amp-story.js directly. This way, no need to call any method on the interactive instance from story.js directly

Reply:
Done! Lmk if the current implementation looks better